common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -276,9 +342,13 @@ public void onComplete(String streamId) throws IOException {
     @Override
     public void onFailure(String streamId, Throwable cause) throws IOException {
       channel.close();
```

Comment:
What is the expected behavior if there are exceptions closing channel ? (the failure perhaps being due to `onData` throwing exception, for example)

Reply:
I reverted back the code. We do want the same behavior, that is, to fail remaining blocks

Reply:
Reverted back this change.